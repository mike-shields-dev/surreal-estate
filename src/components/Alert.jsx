import React from "react";
import PropTypes from "prop-types";
import css from "../styles/Alert.module.css";

const Alert = ({ message, isSuccess }) => {
  if (!message) return null;
  return (
    <div
      className={`
      ${css.Alert} 
      ${css[`${isSuccess ? "success" : "error"}`]}
      `}
      title="alert"
    >
      {message}
    </div>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  isSuccess: PropTypes.bool.isRequired,
};

export default Alert;
