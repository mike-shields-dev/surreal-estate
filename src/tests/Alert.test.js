import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

const successProps = {
  message: "Success!",
  isSuccess: true,
};

const errorProps = {
  message: "Error!",
  isSuccess: false,
};

describe("Alert", () => {
  it("renders the provided message", () => {
    const { asFragment } = render(<Alert {...successProps} />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(successProps.message)).toBeInTheDocument();
  });

  it("renders nothing if no message is provided", () => {
    const { asFragment } = render(<Alert message="" isSuccess />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.queryByTitle("alert")).toBeFalsy();
  });

  it("has the correct class name when given success prop equal to true", () => {
    const { asFragment } = render(<Alert {...successProps} />);

    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByText(successProps.message).getAttribute("class")
    ).toMatch(/success/i);
  });

  it("has the correct class name when given success prop equal to false", () => {
    const { asFragment } = render(<Alert {...errorProps} />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(errorProps.message).getAttribute("class")).toMatch(
      /error/i
    );
  });
});
