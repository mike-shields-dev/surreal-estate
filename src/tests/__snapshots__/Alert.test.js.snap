// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert has the correct class name when given success prop equal to false 1`] = `
<DocumentFragment>
  <div
    class="
      Alert 
      error
      "
    title="alert"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert has the correct class name when given success prop equal to true 1`] = `
<DocumentFragment>
  <div
    class="
      Alert 
      success
      "
    title="alert"
  >
    Success!
  </div>
</DocumentFragment>
`;

exports[`Alert renders nothing if no message is provided 1`] = `<DocumentFragment />`;

exports[`Alert renders the provided message 1`] = `
<DocumentFragment>
  <div
    class="
      Alert 
      success
      "
    title="alert"
  >
    Success!
  </div>
</DocumentFragment>
`;
